apps/service_broadcast/lib/broadcast/init.ex

```diff
@@ -1,16 +1,22 @@
 defmodule Broadcast.Init do
+  use Retry
   use Initializer,
     name: __MODULE__,
     supervisor: Broadcast.Stream.Supervisor
 
   def on_start(state) do
-    Broadcast.Stream.Store.get_all!()
-    |> Enum.reject(&is_nil/1)
-    |> Enum.reject(&Broadcast.Stream.Store.done?(&1))
-    |> Enum.each(fn load ->
-      Broadcast.Stream.Supervisor.start_child({Broadcast.Stream, load: load})
-    end)
-
-    {:ok, state}
+    retry with: constant_backoff(100) do
```

Comment:
have a look at AnnotatedRetry, it does the same thing with mostly the same arguments but in a cleaner way. It's an app within the umbrella.

Reply:
Also, looking at everything that pipe is doing, I'd just create a separate function to retry connecting to the database to "list tables" or similar and retrying only that one. Then make a single call to the `get_all` that pipes through three other things we probably don't want to retry.

Reply:
Thanks Jeff! I didn't realize AnnotatedRetry was a separate thing. Also, does the Brook Postgres driver have a good way to do "list tables" or would that be a case of me running a SQL statement to list the tables? I don't like the 2nd option because it feels like the app may know too many implementation details of the underlying storage layer, but if that's the cleanest way, I can do that.

Reply:
this could be tricky since postgres doesn't really have a way to "list tables" in SQL (the command is something like `/dt`) so that's a limitation of the underlying Postgrex driver. You'll probably need to run the `get_all` query on its own until it returns a success and then send that through the pipeline.

Reply:
I switched to Annotated Retry. Brian is suggesting putting that code in Initializer.ex so all the apps use it. Since every init right now is starting out with getting everything from the store, I think this is fine.